### DISCUSSION TITLE
"nested" es module imports failing

### DISCUSSION THREAD

**Discussion by harrygr** [remix-run/remix#7532] [Q&A]
Having just upgraded my Remix project to v2 I have set `serverModuleFormat` to "esm" so I use ES modules.

I have an import somewhere like:
```
import isValid from "date-fns/isValid";
```

When I build and run my project I get errors like this:

```
Error: Directory import '/path/to/project/node_modules/date-fns/isValid' is not supported resolving ES modules imported from /path/to/project/build/index.js
Did you mean to import date-fns/isValid/index.js?
```

I thought this might be a problem with `date-fns`, but it looks like `date-fns` should support ES Module just fine and has a package.json at `node_modules/date-fns/isValid` that looks like this:

```
{
  "sideEffects": false,
  "module": "../esm/isValid/index.js",
  "typings": "../typings.d.ts"
}
```

If I import `isValid` from the parent module like so it works OK

```
import { isValid } from "date-fns";
```

Any ideas why this is?

**Comment by pointhalo** [comment]
Encountered the same problem when I relied on the locale functionality of `date-fns` v2.29.3

```
import { zhCN } from 'date-fns/locale';
```

❌ in Remix v2
✅ in Remix v1


Hope to get some feasible suggestions
